src/frontend/context/SecurityContext.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from "react";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { ASYNC_KILL_KEY, PASSWORD_KEY } from "../constants";
+import { useEffect } from "react";
+import { useRef } from "react";
+
+//We should be setting the password to null when the password is turned off
+interface SecuritContextType {
+  killState: boolean;
+  setKillState: (killStateValue: boolean) => void;
+  passcode: string | null;
+  setPasscode: (passValue: string | null) => void;
+  checkFlag: boolean;
+  killStateActive: boolean;
+  setKillStateActive: (isActive: boolean) => void;
+  setCheckFlag: (checkFlag: boolean) => void;
+}
+
+const DEFAULT_CONTEXT: SecuritContextType = {
+  killState: false,
+  setKillState: () => {},
+  setPasscode: () => {},
+  passcode: null,
+  checkFlag: false,
+  setCheckFlag: () => {},
+  killStateActive: false,
+  setKillStateActive: () => {},
+};
+
+export const SecurityContext = React.createContext<SecuritContextType>(
+  DEFAULT_CONTEXT
+);
+
+export const SecurityProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [killState, setKillState] = React.useState(false);
+  const [killStateActive, setKillStateActive] = React.useState(false);
```

Comment:
what exactly is the difference between these? is `killState` for allowing the option to exist in the first place and `killStateActive` reflecting whether or not it's actually been enabled by the user? Either way, would update the variable names to make this clearer

at a glance, potentially could simplify it to a single variable

Reply:
So Killstate, is when the user is in Kill mode (aka, no observations, and looks like it is fresh download). KillStateActive indicated that the user CAN go into kill mode. I could reduce this to an object, which might be clearer. 

I do agree these variable names are not the best. Do you have any suggestion for clearer names (im strugging to think of anything)

Reply:
I see. Perhaps the states here can be modified to make this clearer then:

```ts
// Self-explanatory. If it's `null`, the user hasn't set a passcode for the app
// Use this to determine whether the screen kill code toggle is even available in the first place, since that can only be enabled/disabled if the user has a passcode to begin with
const [passcode, setPasscode] = useState<string | null>(null);

// Use a string union to reflect the app mode the user is currently in.
// This allows more flexibility moving forward and is more expressive, requiring fewer boolean state variables
//   - setAppMode('kill'): when user enters kill code when logging in
//   - setAppMode('normal'): when user exists kill mode (however that's done)
const [appMode, setAppMode] = useState<'normal' | 'kill'>('normal');

// Represents the user's choice to allow use of the kill code or not (i.e. the toggle in the settings screen)
// You should only be able to update this if the user has a passcode set
const [killCodeEnabled, setKillCodeEnabled] = useState<boolean>(false)

// A more expressive and flexible alternative to the `checkFlag` state
//   - setAuthStatus('pending'): used when initially loading the app or returning from the background (assuming a passcode has been set)
//   - setAuthStatus('authenticated'): used when the user correctly inputs either their passcode or the kill code
//   - setAuthStatus('notRequired'): used when the user has a passcode set but then decides to disable that feature
const [authStatus, setAuthStatus] = useState<'pending' | 'authenticated' | 'notRequired'>(() => {
  // use `passcode` to determine which initial value this is
  return !!passcode ? 'pending' : 'notRequired'
})
```

Reply:
Whoops sorry, I just edited your comment directly, meant to reply!